pkg/utils/validator/resourceoverride_test.go

```diff
@@ -580,3 +682,100 @@ func TestValidateOverridePolicy(t *testing.T) {
 		})
 	}
 }
+
+func TestValidateJSONPatchOverride(t *testing.T) {
+	tests := map[string]struct {
+		jsonPatchOverrides []placementv1alpha1.JSONPatchOverride
+		wantErrMsg         error
+	}{
+		"valid json override patch": {
+			jsonPatchOverrides: []placementv1alpha1.JSONPatchOverride{
+				{
+					Operator: placementv1alpha1.JSONPatchOverrideOpReplace,
+					Path:     "/spec/clusterResourceSelector/kind",
+					Value:    apiextensionsv1.JSON{Raw: []byte(`"ClusterRole"`)},
+				},
+			},
+			wantErrMsg: nil,
+		},
+		"invalid resource override path - cannot override typeMeta fields (kind)": {
+			jsonPatchOverrides: []placementv1alpha1.JSONPatchOverride{
```

Comment:
For either of those scenarios, those are supposed to pass, right? Along with the scenario where it could be `spec.resourceclusterselectors", would we want to check for periods or wait for the mutation webhook to change that to be correct path? For any of these scenarios, they should be rejected when trying to apply because they are not true paths.

Reply:
For these two scenarios, they're invalid. So we should reject them.

A valid path should begin with "/" based on the rfc6902.

For the "spec.resourceclusterselectors", cx could create a field itself called "spec.resourceclusterselectors", If the path is "/spec.resourceclusterselectors", we could allow and controller will check during the runtime.